src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -222,19 +222,17 @@ Copyright (c) .NET Foundation. All rights reserved.
       <NoWarn>$(NoWarn);IL2092;IL2093;IL2094;IL2095</NoWarn>
       <!-- DynamicallyAccessedMembersAttribute used on unsupported member -->
       <NoWarn>$(NoWarn);IL2097;IL2098;IL2099</NoWarn>
-
-      <!-- Framework has DynamicDependencyAttributes that reference windows-only members. -->
-      <NoWarn Condition=" !$(RuntimeIdentifier.StartsWith('win')) ">$(NoWarn);IL2037</NoWarn> <!-- Unresolved member for DynamicDependencyAttribute -->
-      <!-- Framework embedded XML descriptors reference 32-bit-only members. -->
-      <NoWarn Condition=" '$(PlatformTarget)' != 'x64' AND '$(PlatformTarget)' != 'arm64'">$(NoWarn);IL2009;IL2012</NoWarn> <!-- Unresolved field referenced in XML -->
     </PropertyGroup>
 
-    <PropertyGroup Condition=" '$(PublishTrimmed)' == 'true' ">
-      <!-- Temporarily suppress these warnings regardless of SuppressTrimAnalysisWarnings to remove spurious warnings on non-Windows platforms. -->
-      <!-- Warnings from the framework. Can be removed once https://github.com/dotnet/runtime/issues/40336 is fixed. -->
+    <!-- Suppressions to work around issues in previous versions of the framework. See https://github.com/dotnet/runtime/issues/40336 -->
+    <PropertyGroup Condition="'$(SuppressTrimAnalysisWarnings)' == 'true' And $([MSBuild]::VersionLessThan('$(TargetFrameworkVersion)', '6.0'))">
       <!-- Framework embedded XML descriptors reference windows-only members. -->
       <NoWarn Condition=" !$(RuntimeIdentifier.StartsWith('win')) ">$(NoWarn);IL2008</NoWarn> <!-- Unresolved type referenced in XML. -->
       <NoWarn Condition=" !$(RuntimeIdentifier.StartsWith('win')) ">$(NoWarn);IL2009</NoWarn> <!-- Unresolved member on type referenced in XML. -->
+      <!-- Framework has DynamicDependencyAttributes that reference windows-only members. -->
```

Comment:
Why not fix that instead?

Reply:
These are under a TFM condition - they're removed in .NET6 where we fixed them, but I'm leaving them suppressed for older TFMs since 5.0 shipped with these warnings.